draft/0001-organization.md

```diff
@@ -0,0 +1,292 @@
+  - Throughout this specification, the terms "scope" and "permission" are used interchangeably.
+- **Organization role**: A set of permissions that can be assigned to an organization member.
+- **Organization template**: A collection of organization permissions and roles that applies to every organization.
+
+### 3.2 Limitations
+
+For simplicity, this specification focuses on users as the identity type. Support for other identity types, such as service accounts, can be explored in the future.
+
+## 4. Organization
+
+### 4.1. Definition
+
+An organization is an entity that can connect multiple identities. Each connection is called a membership, and the identity is called an organization member.
+
+### 4.2. Organization template
+
+In modern SaaS applications, it's common to create multiple organizations with the same set of permissions and roles (the template). For instance, a user may have access to multiple workspaces in Slack, each with varying levels of access. In such cases, roles and permissions must be defined within the context of an organization.
+
+While duplicating the template for each organization is feasible, it can become challenging to update the template consistently. For example, adding a new role called "collaborator" would require repeating the same action for every organization. Maintaining consistency across all organizations is crucial.
+
+An organization template is not associated with any specific organization; it is a resource at the tenant level that comprises organization permissions and roles.
+
+> This specification focuses on a single organization template scenario. Multiple organization templates within a tenant are beyond the scope of this discussion.
+
+### 4.3. Organization permission
+
+An organization permission SHOULD be represented as a meaningful string, also serving as the name and unique identifier within the organization template. Permission names should be easily understandable. For example, `read:books` indicates the action to read or retrieve books in the system, while `update:books` signifies updating book information.
+
+It is RECOMMENDED to only use lowercase letters with specific symbols (`:_`) for permissions.
+
+### 4.4. Organization role
+
+An organization role has a unique identifier and contains a set of organization permissions. Each organization role in the template SHOULD only be granted with predefined permissions in the same template. An organization role MAY have no permission granted.
+
+### 4.5. Assign roles to organization members
+
+Before assigning roles to an identity, the identity MUST have a membership with the organization, making them an organization member. An organization member can be assigned multiple roles within the organization, and each role held by the member must be predefined in the organization template. The complete set of permissions granted to an organization member is the union of all permissions granted by their roles.
```

Comment:
Can we directly assign an org role to an identity and hence the identity obtain the membership of the org? If not, what's the reason?

Reply:
no, it's a business need